src/openms/source/FORMAT/HANDLERS/MzMLSpectrumDecoder.cpp

```diff
@@ -306,6 +306,7 @@ namespace OpenMS
 
     if (data.size() > 2) 
     {
+      spectrum.sortByPosition();
```

Comment:
I dont understand how this can work, first you fill in the data in the correct order, then you sort the m/z data only, then you add the meta data? This seems all wrong.